orc8r/cloud/go/obsidian/swagger/spec_servicer_test.go

```diff
@@ -66,6 +76,66 @@ func TestSpecServicer_NewSpecServicerFromFile(t *testing.T) {
 	assert.Equal(t, testStandaloneFileContents, standaloneRes.SwaggerSpec)
 }
 
+func TestSpecServicer_NewSpecServicerFromSpecsPartialErr(t *testing.T) {
+	service := "a service"
+
+	partialErrMock := &mocks.Specs{}
+	partialErrMock.On(
+		"GetPartialSpec", service,
+	).Return(
+		"", errors.New("partial err"),
+	)
+	standaloneSpec := "standalone spec"
+	partialErrMock.On(
+		"GetStandaloneSpec", service,
+	).Return(
+		standaloneSpec, nil,
+	)
+
+	servicer := swagger.NewSpecServicerFromSpecs(partialErrMock, service)
+	assert.NotNil(t, servicer)
+
+	partialRes, err := servicer.GetPartialSpec(
+		context.Background(), &protos.PartialSpecRequest{})
+	assert.NoError(t, err)
+	assert.Empty(t, partialRes.SwaggerSpec)
+
+	standaloneRes, err := servicer.GetStandaloneSpec(
+		context.Background(), &protos.StandaloneSpecRequest{})
+	assert.NoError(t, err)
+	assert.Equal(t, standaloneSpec, standaloneRes.SwaggerSpec)
+}
+
+func TestSpecServicer_NewSpecServicerFromSpecsStandaloneErr(t *testing.T) {
```

Comment:
Looks like your additions remove the need for the two `TestSpecsServicer_*` below, so can remove them

Reply:
While the new tests are covering at the moment, the target unit of the new tests is `NewSpecServicerFromSpecs` and the target of the old tests is `NewSpecServicer`. It would be more complete to leave them all. Also, this ensures coverage is preserved even if the implementation of `NewSpecServicerFromSpecs` changes to not call `NewSpecServicer` internally.

Reply:
Generally my approach to testing non-critical functionality is to be lightweight / ready for change. But I'm fine with leaving it as well, agreed that it does provide a bit of extra completeness